sdk/servicebus/service-bus/src/core/messageSender.ts

```diff
@@ -668,6 +673,94 @@ export class MessageSender extends LinkEntity {
     }
   }
 
+  /**
+   * Returns maximum message size on the AMQP sender link.
+   *
+   * @param {{
+   *       retryOptions?: RetryOptions;
+   *     }} [options={}]
+   * @returns {Promise<number>}
+   * @memberof MessageSender
+   */
+  async getMaxMessageSize(
+    options: {
+      retryOptions?: RetryOptions;
+    } = {}
+  ): Promise<number> {
+    const retryOptions = options.retryOptions || {};
+    if (this.isOpen()) {
+      return this._sender!.maxMessageSize;
+    }
+    return new Promise<number>(async (resolve, reject) => {
+      try {
+        const senderOptions = this._createSenderOptions(Constants.defaultOperationTimeoutInMs);
+        await defaultLock.acquire(this.senderLock, () => {
+          const config: RetryConfig<void> = {
+            operation: () => this._init(senderOptions),
+            connectionId: this._context.namespace.connectionId,
+            operationType: RetryOperationType.senderLink,
+            retryOptions: retryOptions
+          };
+
+          return retry<void>(config);
+        });
+        resolve(this._sender!.maxMessageSize);
+      } catch (err) {
+        reject(err);
+      }
+    });
+  }
+
+  async createBatch(options?: CreateBatchOptions): Promise<ServiceBusMessageBatch> {
+    throwErrorIfConnectionClosed(this._context.namespace);
+    if (!options) {
+      options = {};
+    }
+    let maxMessageSize = await this.getMaxMessageSize();
```

Comment:
Shouldn't we pass the retry options to `getMaxMessageSize()` here?

Reply:
abortSignal and the tracing options too, right?
Added retryOptions to CreateBatchOptions and passing them along to getMaxMessageSize for now.

Like I mentioned here, https://github.com/Azure/azure-sdk-for-js/pull/7527#discussion_r394132920, once we take care of passing the option bags to internal methods across the API, everything would be consistent.